operators.js

```diff
@@ -195,45 +205,83 @@ function paginate(pageSize) {
   return (ops) => updateMeta(ops, 'pageSize', pageSize);
 }
 
+async function executeDeferredOps(ops, meta) {
+  // Collect all deferred tasks and their object-traversal paths
+  const allDeferred = [];
+  traverse(ops).forEach(function (val) {
+    if (!val) return;
+    // this.block() means don't traverse inside these, they won't have DEFERRED
+    if (this.path.length === 1 && this.key === 'meta') return this.block()
+    if (val.type === 'Buffer' && Array.isArray(val.data)) return this.block();
```

Comment:
Can we make this more general maybe? The problem is if someone adds a new operator they will most likely forget to add it here.

Reply:
Apart from the Buffer check (which I think we should keep), I could also do the inverse logic: look only for AND, OR and DEFERRED. I think that's enough to look for the all the depths of the object?

Reply:
Oh and by the way, if someone forgets to update this list when a new operation is added, everything will still just work. This list is just an optimization. We could remove all of these checks and it would still work. I just wanted to remove some unnecessary calculations, even though they aren't super heavy calculations.

Reply:
Yeah only looking for those I think would be better.